signal/src/lib.rs

```diff
@@ -226,7 +226,7 @@ pub mod pallet {
 	pub enum Error<T> {
 		AuthorizationError,
 		BalanceLow,
-		CampaignFailed,
+		CampaignNotSucceeded,
```

Comment:
why rename? also there are words like failed, unsuccessful as part of the english language which are better understandable, also across documentation etc... wdyt?

Reply:
The error name reflects how the check is implemented.
`CampaignFailed` - campaign's state is exactly Failed.
`CampaignNotSucceeded` - campaign's state is one of: Created, Activated, Paused, Failed, Locked. And not Succeeded.

And this is the check under the error:
```
	fn is_campaign_succeeded(campaign_id: &T::Hash) -> bool {
		CampaignStates::<T>::get(campaign_id) == CampaignState::Succeeded
	}
```

Reply:
`!= CampaignState::Succeeded` ?

Reply:
This doesn't make sense to me. The check is used for the [Withdrawal proposal creation ]( https://github.com/gamedaoco/gamedao-protocol/blob/release-1.1.2/signal/src/lib.rs#L313)if the campaign succeeded or not.
So, if the status is other than `Succeeded`, it should be a descriptive error, and `CampaignFailed` doesn't properly describe it.
@2075, what's your suggestion?